servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyHttpServer.java

```diff
@@ -266,7 +266,8 @@ public String toString() {
             this.splittingFlushStrategy = new SplittingFlushStrategy(connection.defaultFlushStrategy(),
                     itemWritten -> {
                         if (itemWritten instanceof HttpResponseMetaData) {
-                            return Start;
+                            final HttpResponseMetaData metadata = (HttpResponseMetaData) itemWritten;
+                            return emptyMessageBody(metadata) && metadata.version().major() > 1 ? End : Start;
```

Comment:
@Scottmitch I understand that a different format of the flattened HTTP message needs additional coordination for places that expect trailers at the end. However, the perf bump of reducing the number of h2 frames is significant enough (see description of #1518) to justify these adjustments. Also, I hope that this `SplittingFlushStrategy` logic will go away after we rework the server-side internals.

Reply:
I wasn't able to comment on this during review, so I moved this down to here: https://github.com/apple/servicetalk/pull/1679#discussion_r672732144